packages/core-data/src/use-entity-mutation.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Internal dependencies
+ */
+import { useMemo } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { store as coreStore } from '../';
+
+export default function useEntityMutation( kind, type, id ) {
+	const mutationState = useSelect(
+		( select ) => {
+			const args = [kind, type, id];
+			return {
+				editedRecord: select( coreStore ).getEditedEntityRecord( ...args ),
+				hasEdits: select( coreStore ).hasEditsForEntityRecord( ...args ),
+				isSaving: select( coreStore ).isSavingEntityRecord( ...args ),
+				saveError: select( coreStore ).getLastEntitySaveError( ...args ),
+				isDeleting: select( coreStore ).isDeletingEntityRecord( ...args ),
+				deleteError: select( coreStore ).getLastEntityDeleteError( ...args ),
+			};
+		},
+		[kind, type, id],
+	);
+
+	const {
+		editEntityRecord,
+		saveEditedEntityRecord,
+		deleteEntityRecord,
+	} = useDispatch( coreStore );
+
+	const mutationFunctions = useMemo(
+		() => ( {
+			edit: ( record ) => editEntityRecord( kind, type, id, record ),
+			save: ( record ) =>
+				saveEntityRecord( kind, type, id, record ),
+			saveEdits: () => saveEditedEntityRecord( kind, type, id ),
+			delete: () => deleteEntityRecord( kind, type, id ),
+		} ),
+		[id],
+	);
+
+	return {
+		...mutationFunctions,
```

Comment:
Looking at the resulting API I'm wondering whether it might make for more readable components if we instead expose an object on a `mutations` property. 

That would mean we'd call `mutations.edit()` rather than just `edit()`.

Just an idea.

Reply:
There's much less data coming out of this hook now – do you think that removes the need for the additional nesting level of `edit`?